### DISCUSSION TITLE
How to create a prov. profile from a local .p12 on Spaceship?

### DISCUSSION THREAD

**Discussion by elpa-shazito** [fastlane/fastlane#16838] [Q&A]
Hi

I'm wondering if I can use Spaceship to create a new provisioning profile using a local .p12 not on keychain.

I see on [Spaceship docs to create a profile](https://rdoc.info/gems/fastlane/Spaceship/Portal/ProvisioningProfile#create!-class_method) that you can provide the `cert` param, but I cannot find a way to use a local .p12 to do it. 

I understand that I cannot use a cert downloaded directly from Apple Developer Portal to create a profile if I dont have it installed, since it doesn't have a private key. On [Spaceship example ](https://github.com/fastlane/fastlane/tree/master/spaceship#example-spaceship-code) cert is already installed?

Also do not know if this is its correct use but I tried to read the p12 and create the `cert` instance using the [cert factory method](https://rdoc.info/gems/fastlane/Spaceship/Portal/Certificate#factory-class_method) but that doesn't seems to work throwing `[]=': string not matched (IndexError)`

```ruby
f = File.read("cert.cer")
cert = Spaceship.certificate.factory(f)
```

The reason I want to use spaceship instead of "regular" fastlane is because it matters in this use case the execution time that `sigh` takes

---
Thanks for all the great work u put in  :)

**Comment by joshdholtz** [comment]
@elpa-shazito Hey 👋 This is great question 😊 

So... the provisioning profile isn't created on the the client side. You don't need the physical .p12 to create the profile. The provisioning profile is created with the ID of the certificate (where the ID is the value associated with the certificate on Apple's ID).

I have a big change that adds the ability to create provisioning profiles with `Spaceship::ConnectAPI` - https://github.com/fastlane/fastlane/pull/16158/files#diff-e9e7336e24f66a051f7c7e07128efd8fR194-R199

You can see here the profile is simply being created with a certificate id (not a physical p12 on the machine). You should be able to do something like this manually next week when its release 😇

**Reply by aschenkel** [reply]
Great! Thanks a lot @joshdholtz , that's clarifying
